elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/AggregationDataStoreHelper.java

```diff
@@ -15,43 +21,75 @@
 import java.util.LinkedHashMap;
 import java.util.LinkedHashSet;
 import java.util.Map;
-import java.util.Optional;
 import java.util.Set;
 
+/**
+ * Helper for Aggregation Data Store which does the work associated with extracting {@link Query}.
+ */
 public class AggregationDataStoreHelper {
 
+    private static final int AGGREGATION_METHOD_INDEX = 0;
+
     private Schema schema;
     private EntityProjection entityProjection;
+    private Set<Dimension> dimensions;
+    private Set<TimeDimension> timeDimensions;
+    private Map<Metric, Class<? extends Aggregation>> metricMap;
+    private FilterExpression whereFilter;
+    private FilterExpression havingFilter;
 
-    public AggregationDataStoreHelper(Schema schema, EntityProjection entityProjection) {
+    public AggregationDataStoreHelper(Schema schema, RequestScope scope) {
         this.schema = schema;
-        this.entityProjection = entityProjection;
+        this.entityProjection = scope.getEntityProjection();
+        dimensions = new LinkedHashSet<>();
+        timeDimensions = new LinkedHashSet<>();
+        getDimensionLists(dimensions, timeDimensions);
+        Set<Metric> metrics = new LinkedHashSet<>();
+        getMetricList(metrics);
+        getMetricMap(metrics);
+        getFilters(entityProjection.getFilterExpression());
     }
 
-    public FilterExpression getWhereFilter(FilterExpression filterExpression) {
-        FilterExpression whereFilter;
-        try {
-            whereFilter = filterExpression.accept(new SplitFilterExpressionVisitor(schema)).getWhereExpression();
-        } catch (NullPointerException npe) {
-            whereFilter = null;
-        }
-        return whereFilter;
+    /**
+     * @return {@link Query} query object with all the parameters provided by user.
+     */
+    public Query getQuery() {
+        return Query.builder()
+                .schema(schema)
+                .metrics(metricMap)
+                .groupDimensions(dimensions)
+                .timeDimensions(timeDimensions)
+                .whereFilter(whereFilter)
+                .havingFilter(havingFilter)
+                .sorting(entityProjection.getSorting())
+                .pagination(entityProjection.getPagination())
+                .build();
     }
 
-    public FilterExpression getHavingFilter(FilterExpression filterExpression) {
-        FilterExpression havingFilter;
-        try {
-            havingFilter = filterExpression.accept(new SplitFilterExpressionVisitor(schema)).getHavingExpression();
-        } catch (NullPointerException npe) {
+    /**
+     * Gets whereFilter and havingFilter based on provided filter expression from {@link EntityProjection}.
+     * @param filterExpression {@link FilterExpression} The filter expression to parse.
+     */
+    private void getFilters(FilterExpression filterExpression) {
```

Comment:
Do we need to pass in a FilterExpression or we can just use the private variable `entityProjection`. And as this method is not returning anything, we can named it `resolveFilters` or `splitFilters`, similar for other methods in this class.